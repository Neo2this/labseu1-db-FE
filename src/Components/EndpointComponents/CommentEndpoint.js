/*
For each comment:

- [x]  Comment body
- [x]  Date comment created
- [x]  Full name of user who created it
- [x]  User profile picture
- [x]  Array of ids of people who liked it
- [x]  Decision Boolean
*/

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { firestoreConnect } from 'react-redux-firebase';

class CommentEndpoint extends Component {
  render() {
    return (
      <div>
        <h2>Comment Endpoint</h2>
        {this.props.comments.map(c => {
          return (
            <div key={c.id}>
              <div>Comment body: {c.commentBody}</div>
              <div>Comment created at: {c.commentCreatedAt.seconds}</div>
              <div>Comment created by: {c.commentCreatedByUserName}</div>
              <div>Comment created by: {c.commentCreatedByUserName}</div>
              <div>Is comments decision: {c.isCommentDecided}</div>
              {c.arrayOfUserIdsWhoLiked.map(u => {
                return <div key={u}>Ids of users who liked the comment: {u}</div>;
              })}
            </div>
          );
        })}
        {this.props.users.map(u => {
          return (
            <div key={u.id}>
              <div>User profile picture: {u.profileUrl}</div>
            </div>
          );
        })}
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    comments: state.firestore.ordered.comments ? state.firestore.ordered.comments : [],
    users: state.firestore.ordered.users ? state.firestore.ordered.users : []
  };
};

const mapDispatchToProps = {};

//Connect to Firestore
export default compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  firestoreConnect(props => {
    return [
      {
        collection: 'comments'
      },
      {
        collection: 'users'
      }
    ];
  })
)(CommentEndpoint);
