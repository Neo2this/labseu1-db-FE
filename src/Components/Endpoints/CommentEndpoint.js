/*
For each comment:

- [x]  Comment body
- [x]  Date comment created
- [x]  Full name of user who created it
- [ ]  User profile picture - WE SHOULD ADD PROFILE PICTURE TO COMMENT OR NOT USE IT
- [x]  Array of ids of people who liked it
- [x]  Decision Boolean
*/

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { firestoreConnect } from 'react-redux-firebase';

class CommentEndpoint extends Component {
  render() {
    console.log(this.props.comment);
    return (
      <div>
        <h2>Comment Endpoint</h2>
        {this.props.comment.commentBody && <div>{this.props.comment.commentBody}</div>}
        {this.props.comment.commentCreatedAt && <div>{this.props.comment.commentCreatedAt.seconds}</div>}
        {this.props.comment.commentCreatedByUserName && <div>{this.props.comment.commentCreatedByUserName}</div>}
        {this.props.comment.isCommendDecided && <div>{this.props.comment.isCommendDecided.toString()}</div>}
        {this.props.comment.arrayOfUserIdsWhoLiked && (
          <div>
            {this.props.comment.arrayOfUserIdsWhoLiked.map(u => {
              return <div key={u}>{u}</div>;
            })}
          </div>
        )}
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    comment: state.firestore.ordered.comments ? state.firestore.ordered.comments[0] : []
  };
};

const mapDispatchToProps = {};

//Connect to Firestore
export default compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  firestoreConnect(props => {
    return [
      {
        collection: 'comments',
        doc: '02233f0c-d49b-4cf5-849e-3d86e962447c'
      }
    ];
  })
)(CommentEndpoint);
