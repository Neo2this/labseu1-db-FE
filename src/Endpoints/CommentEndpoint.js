/* Endpoints for each comment:

- [x]  Comment body
- [x]  Date comment created
- [x]  Full name of user who created it
- [x]  Array of ids of people who liked it
- [x]  Decision Boolean

- [ ]  User profile picture - WE SHOULD ADD USERID TO COMMENT 
*/

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { firestoreConnect } from 'react-redux-firebase';
import styled from 'styled-components';

//Doc uuid() variable for user
const commentDoc = '00ca546f-606a-4e63-9454-9ca621c74502';

class CommentEndpoint extends Component {
  render() {
    const activeComment = this.props.comment;
    return (
      <SDCard>
        <h2>Comment Endpoint</h2>
        <div>
          <SDSpan>Comment body: </SDSpan>
          {activeComment.commentBody && <span>{activeComment.commentBody}</span>}
        </div>
        <div>
          <SDSpan>Comment created at: </SDSpan>
          {activeComment.commentCreatedAt && <span>{activeComment.commentCreatedAt.seconds}</span>}
        </div>
        <div>
          <SDSpan>Comment created by: </SDSpan>
          {activeComment.commentCreatedByUserName && <span>{activeComment.commentCreatedByUserName}</span>}
        </div>
        <div>
          <SDSpan>Comment liked by these user ids: </SDSpan>
          {activeComment.arrayOfUserIdsWhoLiked && (
            <span>
              {activeComment.arrayOfUserIdsWhoLiked.map(id => {
                return <div key={id}>â€¢ {id}</div>;
              })}
            </span>
          )}
        </div>
        <div>
          <SDSpan>Comment is decision: </SDSpan>
          {activeComment.id && <span>{activeComment.isCommentDecided.toString()}</span>}
        </div>
      </SDCard>
    );
  }
}

const mapStateToProps = state => {
  return {
    comment: state.firestore.ordered.comments ? state.firestore.ordered.comments[0] : []
  };
};

//We are not dispatching anything => mapDispatchToProps is empty
const mapDispatchToProps = {};

//Connect to Firestore
export default compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  firestoreConnect(props => {
    return [
      {
        collection: 'comments',
        doc: `${commentDoc}`
      }
    ];
  })
)(CommentEndpoint);

//Styling
const SDCard = styled.div`
  line-height: 2;
  font-family: 'Helvetica';
  margin: 10px;
  padding: 10px;
  background-color: #eaeef7;
  width: 30%;
`;

const SDSpan = styled.span`
  font-weight: bold;
`;
