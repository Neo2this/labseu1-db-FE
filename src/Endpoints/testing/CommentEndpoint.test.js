import React from 'react';
import * as rtl from 'react-testing-library';
import { CommentEndpoint } from '../CommentEndpoint';

const comment = {
  commentBody: 'Apple',
  commentCreatedAt: { seconds: 12345678 },
  commentCreatedByUserName: 'Test Username',
  arrayOfUserIdsWhoLiked: [1, 3, 5],
  isCommentDecided: true
};

afterEach(rtl.cleanup);

describe('Comment Endpoint', () => {
  it('match snapshot', () => {
    const wrap = rtl.render(<CommentEndpoint comment={comment} />);
    expect(wrap.asFragment()).toMatchSnapshot();
  });
});
