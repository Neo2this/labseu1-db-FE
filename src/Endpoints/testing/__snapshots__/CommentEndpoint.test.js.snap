// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Endpoint match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa QUjQF"
  >
    <h2>
      Comment Endpoint
    </h2>
    <div>
      <span
        class="sc-bwzfXH hZEuEv"
      >
        Comment body: 
      </span>
      <span>
        Apple
      </span>
    </div>
    <div>
      <span
        class="sc-bwzfXH hZEuEv"
      >
        Comment created at: 
      </span>
      <span>
        12345678
      </span>
    </div>
    <div>
      <span
        class="sc-bwzfXH hZEuEv"
      >
        Comment created by: 
      </span>
      <span>
        Test Username
      </span>
    </div>
    <div>
      <span
        class="sc-bwzfXH hZEuEv"
      >
        Comment liked by these user ids: 
      </span>
      <span>
        <div>
          • 1
        </div>
        <div>
          • 3
        </div>
        <div>
          • 5
        </div>
      </span>
    </div>
    <div>
      <span
        class="sc-bwzfXH hZEuEv"
      >
        Comment is decision: 
      </span>
    </div>
  </div>
</DocumentFragment>
`;
